src/docker-in-docker/install.sh

```diff
@@ -298,6 +298,45 @@ else
     update-alternatives --set docker-compose /usr/local/bin/compose-switch
 fi
 
+# Create service for starting and stopping docker daemon
+tee /etc/init.d/docker > /dev/null \
+<< 'EOF'
+PID=$(pidof dockerd)
+case "$1" in
+  start)
+    echo "Starting Docker Daemon..."
+    sudo /usr/bin/dockerd > /dev/null 2>&1 &
```

Comment:
Should this invoke the same entrypoint script that we're currently using to start the docker daemon (/usr/local/share/docker-init.sh).  That entrypoint is doing additional work that (at least historically) was important to do for "docker-in-docker".

Reply:
Well, there are two possible ways to implement this feature:

1. Extend the current entrypoint script
Benefit: The configuration options (MTU, IPv6, DNS etc) could be configured once in the `devcontainer.json`
Drawback: For changing any set Docker Deamon variables you have to rebuild your devcontainer. Maybe you are using an expensive post-create script. In this case you might not like to rebuild your container

2. Creating a start-stop-script
Benefit: Due to the fact that it's a development environment it would be way easier to quickly change any setting in `/etc/docker/deamon.json` and restart the Docker Deamon to continue developing without a full rebuild. Also this way of restarting services is well kown under Linus users.

So both ways would be a nice feature addition in my eyes. How should I continue?

Reply:
I think that the way [options](https://github.com/devcontainers/features/tree/main/src/docker-in-docker#options) are currently baked into the `/usr/local/share/docker-init.sh` is OK (the options are written to that file once when building the container based on the options in the `devcontainer.json`)

Would it make sense to just call this script in the `start` so that the options set from the user's dev container are reapplied here?
```suggestion
    sudo  /usr/local/share/docker-init.sh > /dev/null 2>&1 &
```

Also, does the `service start docker` command automatically get run when starting up the container, or is this just available to a user to run manually themselves?

Reply:
The Docker Deamon gets started when the container is built. My current implementation only allows the user to manually restart the deamon.

It really depends on the use-case. I for myself was confused about the missing options to restart a service.

But I can change this also as following:

```
"features": {
		"ghcr.io/devcontainers/features/docker-in-docker:2": {
			"version": "latest",
			"enableNonRootDocker": "true",
			"moby": "true",
			"dockerDashComposeVersion": "v2"
                        "enableIPv6": "true"                           <------------ as an option
		}
	}
```
So you build your devcontainer once and cannot restart the docker deamon manually.

`/usr/local/share/docker-init.sh` currently uses e.g. the `"moby": "true"` option to decide which action to take. A re-running of this init script makes no sense in my eyes, because there are other options that will also run again.

Maybe it would make more sense to implement it in my first way so that you can configure your docker via the `devcontainer.json`. Then you do not have to create `/etc/docker/deamon.json` which saves you headaches too.

Reply:
> Docker Deamon เริ่มทำงานเมื่อสร้างคอนเทนเนอร์ การใช้งานปัจจุบันของฉันอนุญาตให้ผู้ใช้รีสตาร์ทปีศาจด้วยตนเองเท่านั้น
> 
> ขึ้นอยู่กับกรณีการใช้งานจริงๆ สำหรับตัวฉันเองรู้สึกสับสนเกี่ยวกับตัวเลือกที่ขาดหายไปในการเริ่มบริการใหม่
> 
> แต่ฉันสามารถเปลี่ยนสิ่งนี้ได้ดังต่อไปนี้:
> 
> ```
> "features": {
> 		"ghcr.io/devcontainers/features/docker-in-docker:2": {
> 			"version": "latest",
> 			"enableNonRootDocker": "true",
> 			"moby": "true",
> 			"dockerDashComposeVersion": "v2"
>                         "enableIPv6": "true"                           <------------ as an option
> 		}
> 	}
> ```
> 
> ดังนั้นคุณจึงสร้าง devcontainer เพียงครั้งเดียวและไม่สามารถรีสตาร์ท docker deamon ด้วยตนเองได้
> 
> `/usr/local/share/docker-init.sh`ใช้อยู่ในปัจจุบัน เช่น`"moby": "true"`ตัวเลือกในการตัดสินใจว่าจะดำเนินการใด การรันสคริปต์ init ใหม่นี้ไม่สมเหตุสมผลในสายตาของฉัน เพราะมีตัวเลือกอื่นที่จะรันอีกครั้ง
> 
> บางทีมันอาจจะเหมาะสมกว่าที่จะใช้มันในวิธีแรกของฉัน เพื่อให้คุณสามารถกำหนดค่านักเทียบท่าของคุณผ่านไฟล์`devcontainer.json`. จากนั้นคุณไม่จำเป็นต้องสร้าง`/etc/docker/deamon.json`ซึ่งช่วยให้คุณปวดหัวด้วย